File: bindings/node/examples/primitives.mjs
import { backendAvailable, kem, signature, version } from "../dist/index.js";

console.log(`PQC Bridge ${version()}`);
console.log(`ML-KEM-768 available: ${backendAvailable("ML-KEM-768")}`);
console.log(`ML-DSA-65 available: ${backendAvailable("ML-DSA-65")}`);

const kemKeypair = kem.keypair();
const encapsulation = kem.encapsulate(kemKeypair.publicKey);
const decapsulated = kem.decapsulate(
  kemKeypair.secretKey,
  encapsulation.ciphertext,
);

if (!encapsulation.sharedSecret.equals(decapsulated)) {
  throw new Error("ML-KEM shared secrets did not match");
}

const signatureKeypair = signature.keypair();
const message = Buffer.from("pqcb node example", "utf8");
const signatureBytes = signature.sign(signatureKeypair.secretKey, message);
signature.verify(signatureKeypair.publicKey, message, signatureBytes);

console.log("Node primitive example completed");

File: bindings/python/examples/primitives.py
from __future__ import annotations

import sys
from pathlib import Path

sys.path.insert(0, str(Path(__file__).resolve().parents[1]))

import pqcb


print(f"PQC Bridge {pqcb.abi_version()}")
print(f"ML-KEM-768 available: {pqcb.backend_available('ML-KEM-768')}")
print(f"ML-DSA-65 available: {pqcb.backend_available('ML-DSA-65')}")

kem_public_key, kem_secret_key = pqcb.kem_keypair()
ciphertext, encapsulated = pqcb.kem_encapsulate(kem_public_key)
decapsulated = pqcb.kem_decapsulate(kem_secret_key, ciphertext)

if encapsulated != decapsulated:
    raise RuntimeError("ML-KEM shared secrets did not match")

signature_public_key, signature_secret_key = pqcb.signature_keypair()
message = b"pqcb python example"
signature = pqcb.sign(signature_secret_key, message)
pqcb.verify(signature_public_key, message, signature)

print("Python primitive example completed")
